### PR TITLE
Vite 4 + Vue 3.3 + TS 5 support

### PR DESCRIPTION
**Pull request by mrcego**
Based in #2, I was working in Vite 4 + TypeScript 5 + Vue 3.3 support ensuring correct solution at `demo-app-vite`.

Please, don't merge until checklist is completed.

### Checklist

- [x] [unplugin-auto-import](https://github.com/antfu/unplugin-auto-import)
- [x] [unplugin-vue-components](https://github.com/antfu/unplugin-vue-components)
- [x] Preconfigured TypeScript support and `d.ts` environment generation for wrapper.
- [x] Pinia support (+ plugins)
- [x] Form generation with Veevalidate
- [x] Convert to monorepo
- [x] Build test
- [x] `?inline` support for imported styles
- [x] Proper types

### CODE REVIEW

**Comment by mrcego** [comment]
@EranGrin I have marked it as ready to review to listen feedbacks and thoughts from you about current work. Regards.

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (EranGrin) [inline .prettierrc:1]: I'm usually using only ESLint for my vue projects, but if prettier is more common I don't mind

File: package/pnpm-lock.yaml
Comment (EranGrin) [inline package/pnpm-lock.yaml:1]: is it good practice to keep the yarn lock and pnpm.lock?  Or do you suggest moving completely to pnpm

File: package/web-component-util.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  plugins,
+  cssFrameworkStyles,
+  VueDefineCustomElement,
+  h,
+  createApp,
+  getCurrentInstance,
+}) =>
+  VueDefineCustomElement({
+    styles: [cssFrameworkStyles],
+    // styles: [...getStylesRecursively(rootComponent)],
```
Comment (EranGrin) [inline package/web-component-util.ts:52]: Can we remove this part?

File: package/web-component-util.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      throw new Error(
+        'No parent element found, the rootComponent must be wrapped in a HTML element (e.g. <template><div> app content </div></template>)'
+      )
+    }
+    el = el.parentElement
+  }
+  return el
+}
+
+const getCustomElementStyle = (styles) => {
```
Comment (EranGrin) [inline package/web-component-util.ts:31]: Not sure what the usage of `getCustomElementStyle`

**Review by EranGrin — COMMENTED** [review]
Thank you for the PR
I left a few comments,
but one major aspect I didn't check or mentioned is the vite production build.

File: package/pnpm-lock.yaml
Comment (mrcego) [reply package/pnpm-lock.yaml:1]: I suggest `pnpm` due to monorepo with workspaces.

File: package/web-component-util.ts (hunk 2 above)
Comment (mrcego) [reply package/web-component-util.ts:52]: I will remove it in the next commit.

File: package/web-component-util.ts (hunk 3 above)
Comment (mrcego) [reply package/web-component-util.ts:31]: I will remove it in the next commit.

File: .prettierrc (hunk 1 above)
Comment (mrcego) [reply .prettierrc:1]: We could using both, if you don't mind.

**Comment by mrcego** [comment]
> Thank you for the PR
> I left a few comments,
> but one major aspect I didn't check or mentioned is the vite production build.

Yeah, we need to test this in detail. Even if the build config from this PR works as expected, it would be good if you check it more intense.

**Comment by mrcego** [comment]
@baiwusanyu-c comes (with this gem)[https://github.com/baiwusanyu-c/unplugin-vue-ce/tree/master/packages/ce-app] inspired by this solution.

**Comment by EranGrin** [comment]
Merge PR into feature [branch](https://github.com/EranGrin/vue-web-component-wrapper/tree/feature/vite-support) vite-support

**Pull request merged**